sdks/python/apache_beam/io/parquetio.py

```diff
@@ -0,0 +1,349 @@
+    else:
+      return first_column_metadata.data_page_offset
+
+  @staticmethod
+  def get_number_of_row_groups(pf):
+    return pf.metadata.num_row_groups
+
+
+class _ParquetSource(filebasedsource.FileBasedSource):
+  """A source for reading Parquet files.
+  """
+  def __init__(self, file_pattern, min_bundle_size, validate, columns):
+    super(_ParquetSource, self).__init__(
+        file_pattern=file_pattern,
+        min_bundle_size=min_bundle_size,
+        validate=validate
+    )
+    self._columns = columns
+
+  def read_records(self, file_name, range_tracker):
+    next_block_start = -1
+
+    def split_points_unclaimed(stop_position):
+      if next_block_start >= stop_position:
+        # Next block starts at or after the suggested stop position. Hence
+        # there will not be split points to be claimed for the range ending at
+        # suggested stop position.
+        return 0
+      return RangeTracker.SPLIT_POINTS_UNKNOWN
+
+    range_tracker.set_split_points_unclaimed_callback(split_points_unclaimed)
+
+    start_offset = range_tracker.start_position()
+    if start_offset is None:
+      start_offset = 0
+
+    with self.open_file(file_name) as f:
+      pf = ParquetFile(f)
+
+      index = _ParquetUtils.find_first_row_group_index(pf, start_offset)
+      if index != -1:
+        next_block_start = _ParquetUtils.get_offset(pf, index)
+      else:
+        next_block_start = range_tracker.stop_position()
+      number_of_row_groups = _ParquetUtils.get_number_of_row_groups(pf)
+
+      while range_tracker.try_claim(next_block_start):
+        table = pf.read_row_group(index, self._columns)
+
+        if index + 1 < number_of_row_groups:
+          index = index + 1
+          next_block_start = _ParquetUtils.get_offset(pf, index)
+        else:
+          next_block_start = range_tracker.stop_position()
+
+        num_rows = table.num_rows
+        data_items = table.to_pydict().items()
+        for n in range(num_rows):
+          data = {}
+          for label, d in data_items:
+            data[label] = d[n]
+          yield data
+
+
+class WriteToParquet(PTransform):
+  """A ``PTransform`` for writing parquet files."""
+
+  def __init__(self,
+               file_path_prefix,
+               schema,
+               codec='none',
+               row_group_size=1000,
+               file_name_suffix='',
+               num_shards=0,
+               shard_name_template=None,
+               mime_type='application/x-parquet'):
+    """Initialize a WriteToParquet transform.
+
+    Args:
+      file_path_prefix: The file path to write to. The files written will begin
+        with this prefix, followed by a shard identifier (see num_shards), and
+        end in a common extension, if given by file_name_suffix. In most cases,
+        only this argument is specified and num_shards, shard_name_template, and
+        file_name_suffix use default values.
+      schema: The schema to use, as type of pyarrow.Schema
```

Comment:
Probably give more details and a example for building the schema somewhere in this doc.